src/app/slotting/edit/components/slot.component.ts

```diff
@@ -57,6 +62,18 @@ export class SlotComponent implements OnInit, AfterViewInit {
     if (this.slot.newSlot) {
       this.newSlot = true;
     }
+
+    // check if description is from a preset
+    const desc = this.slot.description.toLowerCase();
+    const preset = this.editService.slotPresets.find(p => p.description.toLowerCase() === desc);
+    if (preset !== undefined) {
```

Comment:
mich gruselt's immer vor direkten Vergleichen mit `undefined`, aber vllt bin ich auch einfach zu alt :D

Reply:
Ja bist du 😉 
> **Return Value**
> The value of the first element in the array that satisfies the provided testing function. Otherwise, undefined is returned.

\- [MDN web docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find#Return_value)

Reply:
/shrug als ich jung war, 

a) konnte man undefined noch zuweisen… und das wäre dann unschön gewesen
b) galt impliziter cast auf boolean ( `if (x)` statt `if (x === undefined)` ) als die sichere Vorgehensweise, weil man nie wußte ob irgendwas undefined oder null zurückgibt

Reply:
Ich mische mich mal random ein 🐰 

Da `preset` nur ein Objekt oder _undefined_ sein kann, ist der strikte Vergleich zu `undefined` etwas overkill, aber eine gute best practise; sollte sich später im Code etwas ändern.

a) Undefined kann man noch immer zuweisen @Fusselwurm was die find Methode ja auch macht, sollte sie kein passendes Element finden.  

b) In Javascript ist ´if(x)´ ein loser Vergleich und ´if(x === undefined)´ ein starker Vergleich. Eine übersichtliche Tabelle gibt es [hier](https://dorey.github.io/JavaScript-Equality-Table/)  

c) Stimme ich dir zu; man muss abwägen was in diesem Fall wichtiger ist Schönheit oder Robustheit für spätere Änderungen.

Reply:
@Frozen-Byte nja ich meine früher konntste `undefined = true`;  sagen, und dann war undefined halt nur ne variable wie jede andere - darum war ein check auf `undefined`nicht ungefährlich

Reply:
Ah, das geht inzwischen nicht mehr, das ist korrekt.

Reply:
Ich habs mir angewöhnt nach nem `array.find` immer ein strikten Vergleich mit dem `undefined` zu machen, da ich so unabhängig von Datentyp der Array Elemente sicher sein kann, dass etwas/nichts gefunden wurde. 

Finde die Robustheit ist da eindeutig wichtiger, und würde es so lassen. Alle Mann zufrieden?